node/src/components/storage/lmdb_store.rs

```diff
@@ -97,6 +97,13 @@ impl<V: Value, M: Send + Sync> Store for LmdbStore<V, M> {
         let serialized_value =
             bincode::serialize(&value).map_err(|error| Error::from_serialization(*error))?;
         let mut txn = self.env.begin_rw_txn().expect("should create rw txn");
+
+        // TODO: this get() call should be removed when we pass WriteFlags::NO_OVERWRITE as below
```

Comment:
Wouldn't the flag be better performance-wise?

Reply:
Yeah it would, but @Fraser999 mentioned a case where we want to append and overwrite currently, to be unwound in the future, and that case is worked around with WriteFlags::default. Unfortunately I'm unfamiliar with that case.

Reply:
Ah right, there's a comment below but it was hidden by GitHub – we currently store block's signatures together with the block but we shouldn't be doing that. They should be stored separately. Adding signature doesn't change block's hash so it wouldn't change `serialized_id` hence we need ability to overwrite. Thanks for clarification.